packages/gestalt/src/Fieldset.test.js

```diff
@@ -0,0 +1,26 @@
+// @flow strict
+import { create } from 'react-test-renderer';
+import Fieldset from './Fieldset.js';
+import Checkbox from './Checkbox.js';
+
+describe('Fieldset', () => {
+  it('renders', () => {
+    const tree = create(
+      <Fieldset legend="What is your favorite dog?">
+        <Checkbox id="Schnauzer" label="Schnauzer" onChange={() => {}} />
+        <Checkbox id="Aussie" label="Aussie" onChange={() => {}} />
+      </Fieldset>,
+    ).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  it('renders a visually hidden legend', () => {
+    const tree = create(
+      <Fieldset legend="What is your favorite dog?" legendDisplay="hidden">
+        <Checkbox id="Schnauzer" label="Schnauzer" onChange={() => {}} />
+        <Checkbox id="Aussie" label="Aussie" onChange={() => {}} />
+      </Fieldset>,
+    ).toJSON();
+    expect(tree).toMatchSnapshot();
```

Comment:
could we do a test that checks that finds (What is your favorite dog?) and returns 1 when visible and 0 when hidden?

Reply:
Since the legend is only hidden with css, the test would just be looking for the visuallyHidden class name, which is essentially the snapshot test anyway so I think it's probably fine. This is where Visual Regression Testing would be ideal instead